### PR TITLE
oesgx to return total EPC size

### PR DESCRIPTION
**Pull request by dmitsh**

### CODE REVIEW

File: tools/oesgx/oesgx.c [hunk 1]
```diff
@@ -128,5 +130,26 @@ int main(int argc, const char* argv[])
...
+            printf("Read SGX_CAPABILITY_ENUMERATION failed:\n");
+            dump_regs(&regs);
+            return result;
+        }
+        if (!HAVE_EPC(regs)) {
+            printf("No EPC section\n");
+            dump_regs(&regs);
+            return 0;
+        }
+        epcSz = ((regs.ecx & 0x0fffff000) | ( (uint64_t)(regs.edx & 0x0fffff) << 32));
```
Comment (anitagov) [inline tools/oesgx/oesgx.c]: Remove the extra space here.   ( (uint64_t)

File: tools/oesgx/oesgx.c [hunk 2]
```diff
@@ -128,5 +130,26 @@ int main(int argc, const char* argv[])
...
+        printf("EPC size=%lu\n", epcSz);
```
Comment (anitagov) [inline tools/oesgx/oesgx.c]: How about adding that this is the EPC size on the platform? 
Change **EPC size** to **EPC size on the platform**

File: tools/oesgx/oesgx.c [hunk 3]
```diff
@@ -128,5 +130,26 @@ int main(int argc, const char* argv[])
         }
         printf("MaxEnclaveSize_64: 2^(%d)\n", (regs.edx >> 8) & 0xFF);
     }
+
+    /* Enumeration of Intel SGX Capabilities: figure out EPC size */
+    {
+        Regs regs = {SGX_CAPABILITY_ENUMERATION, 0, 0x2, 0};
+        uint64_t epcSz = 0;
```
Comment (CodeMonkeyLeet) [inline tools/oesgx/oesgx.c]: Consider `epc_size` for consistency with coding style.

File: tools/oesgx/oesgx.c [hunk 4]
```diff
@@ -12,6 +13,7 @@
 #define HAVE_FLC(regs) (((regs.ecx) >> 30) & 1)
 #define HAVE_SGX1(regs) (((regs.eax) & 1))
 #define HAVE_SGX2(regs) (((regs.eax) >> 1) & 1)
+#define HAVE_EPC(regs) (((regs.eax) & 0x0f) == 0x01)
```
[the remark this answers is not in the document]
Comment (dmitsh) [reply tools/oesgx/oesgx.c]: 1. Changed to HAVE_EPC_SUBLEAF. That makes sense, thanks!
2. Per documentation `EAX Bit 03 - 00: Sub-leaf Type 0000b: Indicates this sub-leaf is invalid. 0001b: This sub-leaf enumerates an EPC section.`  I need to check first 4 bits of EAX.

**Pull request merged**